outport/process/outportDataProvider.go

```diff
@@ -150,41 +175,201 @@ func (odp *outportDataProvider) getSignersIndexes(header data.HeaderHandler) ([]
+		Rewards:              rewards,
+		InvalidTxs:           invalidTxs,
+		Receipts:             receipts,
+		Logs:                 logs,
+	}, nil
+}
+
+func (odp *outportDataProvider) createPoolForMeta(rewardsTxs map[string]data.TransactionHandler) (*outportcore.TransactionPool, error) {
+	txs, err := getTxs(odp.txCoordinator.GetAllCurrentUsedTxs(block.TxBlock))
+	if err != nil {
+		return nil, err
+	}
+
+	scrs, err := getScrs(odp.txCoordinator.GetAllCurrentUsedTxs(block.SmartContractResultBlock))
+	if err != nil {
+		return nil, err
+	}
+
+	rewards, err := getRewards(rewardsTxs)
+	if err != nil {
+		return nil, err
+	}
+
+	logs, err := getLogs(odp.txCoordinator.GetAllCurrentLogs())
+	if err != nil {
+		return nil, err
+	}
+
+	return &outportcore.TransactionPool{
+		Transactions:         txs,
+		SmartContractResults: scrs,
+		Rewards:              rewards,
+		Logs:                 logs,
+	}, nil
+}
+
+func getTxs(txs map[string]data.TransactionHandler) (map[string]*outportcore.TxInfo, error) {
+	ret := make(map[string]*outportcore.TxInfo, len(txs))
+
+	for txHash, txHandler := range txs {
+		tx, castOk := txHandler.(*transaction.Transaction)
+		txHashHex := getHexEncodedHash(txHash)
+		if !castOk {
+			return nil, fmt.Errorf("%w, hash: %s", errCannotCastTransaction, txHashHex)
+		}
+
+		ret[txHashHex] = &outportcore.TxInfo{
+			Transaction: tx,
+			FeeInfo:     newFeeInfo(),
+		}
 	}
+
+	return ret, nil
+}
+
+func getHexEncodedHash(txHash string) string {
+	txHashBytes := []byte(txHash)
+	return hex.EncodeToString(txHashBytes)
 }
 
-func (odp *outportDataProvider) createPoolForMeta(rewardsTxs map[string]data.TransactionHandler) *outportcore.Pool {
-	return &outportcore.Pool{
-		Txs:     WrapTxsMap(odp.txCoordinator.GetAllCurrentUsedTxs(block.TxBlock)),
-		Scrs:    WrapTxsMap(odp.txCoordinator.GetAllCurrentUsedTxs(block.SmartContractResultBlock)),
-		Rewards: WrapTxsMap(rewardsTxs),
-		Logs:    odp.txCoordinator.GetAllCurrentLogs(),
+func newFeeInfo() *outportcore.FeeInfo {
+	return &outportcore.FeeInfo{
+		GasUsed:        0,
+		Fee:            big.NewInt(0),
+		InitialPaidFee: big.NewInt(0),
 	}
 }
 
-func WrapTxsMap(txs map[string]data.TransactionHandler) map[string]data.TransactionHandlerWithGasUsedAndFee {
-	newMap := make(map[string]data.TransactionHandlerWithGasUsedAndFee, len(txs))
-	for txHash, tx := range txs {
-		newMap[txHash] = outportcore.NewTransactionHandlerWithGasAndFee(tx, 0, big.NewInt(0))
+func getScrs(scrs map[string]data.TransactionHandler) (map[string]*outportcore.SCRInfo, error) {
+	ret := make(map[string]*outportcore.SCRInfo, len(scrs))
+
+	for scrHash, txHandler := range scrs {
+		scr, castOk := txHandler.(*smartContractResult.SmartContractResult)
+		scrHashHex := getHexEncodedHash(scrHash)
+		if !castOk {
+			return nil, fmt.Errorf("%w, hash: %s", errCannotCastSCR, scrHashHex)
+		}
+
+		ret[scrHashHex] = &outportcore.SCRInfo{
+			SmartContractResult: scr,
+			FeeInfo:             newFeeInfo(),
```

Comment:
question: FeeInfo will be set later on with the correct values, right?

Reply:
Yes, confirmed it with @miiu96 . It is set in `transactionsFeeProcessor`